### PR TITLE
Fix #36 - shorten the unique ids

### LINKED ISSUE #36: Shorten the ids used for headings
We don't need to be ridiculously unique like a uuid - just mostly unique, or unique within a document

### PR DESCRIPTION
**Pull request by caffodian**
8 chars, alphanumeric 

New id is like:

`autoid-gkb5mz73`

### CODE REVIEW

**Comment by caffodian** [comment]
This test is probably obnoxious.  a 10000 heading document just makes ckeditor soft hang

File: plugins/autoid/plugin.js [hunk 1]
```diff
@@ -85,8 +87,28 @@
...
       function createAutoId() {
-        return 'autoid-' + CKEDITOR.tools.getUniqueId();
+        // adapted from http://fiznool.com/blog/2014/11/16/short-id-generation-in-javascript/
+        // who did a great job on characters to exclude, etc.
+        var uid;
+        var existingIds = findHeadingIds(findAllHeadings(editor));
+
+        while (!uid) {
+          uid = generateUniqueId();
+          if (existingIds.indexOf(uid) !== -1) {
```
Comment (caffodian) [inline plugins/autoid/plugin.js:106]: This should probably never happen.  but hey pseudorandomness

File: plugins/autoid/plugin.js [hunk 2]
```diff
@@ -1,6 +1,8 @@
 'use strict';
 
 (function () {
+  var ALPHABET = '23456789abdegjkmnpqrvwxyz';
```
Comment (caffodian) [inline plugins/autoid/plugin.js:4]: see the reffed blogpost, but this basically eliminates the possibility of ambiguous characters and accidental bad words

File: plugins/autoid/plugin.js [hunk 3]
```diff
@@ -85,8 +87,28 @@
         editor.fire(EVENT_NAMES.ID_ADDED);
       }
 
+      function generateUniqueId() {
+          var uniqueId = '';
+          for (var i = 0; i < ID_LENGTH; i++) {
+            uniqueId += ALPHABET.charAt(Math.floor(Math.random() * ALPHABET.length));
```
Comment (kylegibson-rldatix) [inline plugins/autoid/plugin.js:93]: Is this concatenation method of string construction better/faster than appending to a list and then joining the characters together at the end?

File: plugins/autoid/plugin.js (hunk 2 above)
Comment (kylegibson-rldatix) [reply plugins/autoid/plugin.js:4]: I know you link the blog post below, but maybe put a comment in the code explaining why there appear to be letters missing?

**Comment by kylegibson-rldatix** [comment]
:+1:

**Pull request merged**